traits/tests/test_expression.py

```diff
@@ -53,3 +64,74 @@ def _bar_default(self):
         msg = "The 'bar' trait of a Foo instance must be a valid"
         with self.assertRaisesRegex(TraitError, msg):
             f.bar
+
+    @unittest.expectedFailure  # FIXME issue #1096
+    def test_default_static_override_static(self):
+        class BaseFoo(HasTraits):
+            # The default value set in the class definition is "0"
+            bar = Expression()
+
+        class Foo(BaseFoo):
+            bar = "3"
+
+        f = Foo()
+        self.assertEqual(f.bar, "3")
+        self.assertEqual(f.bar_, compile("3", "<string>", "eval"))
```

Comment:
I'm quite surprised that it's possible to compare code objects for equality this way. If we didn't want to rely on this, it could make sense to test for behaviour instead of exact equality. (i.e., we don't much care what `f.bar_` actually is; we only care that we can `eval` it in a suitable context and get the expected results).

Reply:
Replaced